app/src/services/net/http.service.js

```diff
@@ -122,32 +122,40 @@ export class HttpServiceProvider {
   }
 
   static urlEncode(data) {
+    return `?${HttpServiceProvider.formEncode(data)}`;
+  }
+
+  static formEncode(data) {
     let ret = '';
     for (const key in data) {
       if (ret !== '') {
         ret += '&';
       }
       ret += `${encodeURIComponent(key)}=${encodeURIComponent(data[key])}`;
     }
-    return `?${ret}`;
+    return `${ret}`;
   }
   /** Performs a post request
    * @param {string} url
    * @param {params} {key: value}
    * @param {boolean} urlEncoded
    * @return Observable<{}>
    */
-  post(url, body, urlEncoded, usetoken=true) {
+  post(url, body, urlEncoded, dataInBody = false, usetoken = true) {
     let pUrl = url;
     let pBody = body;
     const headers = new Headers();
 
     headers.set('Content-Type', 'application/json');
 
     if (urlEncoded) {
-      pUrl += HttpServiceProvider.urlEncode(pBody);
       headers.set('Content-Type', 'application/x-www-form-urlencoded; charset=UTF-8');
-      pBody = null;
+      if (dataInBody) {
```

Comment:
Super nitpick, but I think the same function could be used for both, just adding a leading `?` for when it's used in an URL.. :D A function for adding a "?" is a bit overkill. Also I'd keep the name urlencode instead of formencode.

Reply:
Absolutely! I'll fix it right away!

Reply:
Something like facef91 you meant?